crates/services/src/service.rs

```diff
@@ -101,16 +101,25 @@ pub trait RunnableTask: Send {
 #[derive(Debug)]
 pub struct ServiceRunner<S>
 where
-    S: RunnableService,
+    S: RunnableService + 'static,
 {
     /// The shared state of the service
     pub shared: S::SharedData,
     state: Shared<watch::Sender<State>>,
 }
 
+impl<S> Drop for ServiceRunner<S>
+where
+    S: RunnableService + 'static,
+{
+    fn drop(&mut self) {
+        self.stop();
```

Comment:
wouldn't this only help if the service doesn't own the join handles of any sub-tasks it spawns? Or will dropping join handles still allow tspawned tasks to continue executing?

Reply:
> A JoinHandle detaches the associated task when it is dropped, which means that there is no longer any handle to the task, and no way to join on it.

https://docs.rs/tokio/latest/tokio/task/struct.JoinHandle.html

Looks like the spawned tasks just get orphaned and continue to run if the owner drops the join handles.

Reply:
Yep

Reply:
hmm looks like this caused a bunch of test failures

Reply:
Yep because we create a clone of `TxPool` service during each GraphQL request with transactions=)))))

Reply:
Yeh probably needs to be within an Arc?

Reply:
I'm thinking we probably should allow clonable objects to drop inner resources, to avoid any confusion in the future about which resources are really cloneable. Something like an Arc makes sense.